src/object/object_primitive.c

```diff
@@ -8305,11 +8320,13 @@ pr_midxkey_compare_element (char *mem1, char *mem2, TP_DOMAIN * dom1, TP_DOMAIN
   if (!DB_IS_NULL (&val1) && val1.need_clear == true)
```

Comment:
I think it should be:
   if (!DB_IS_NULL (&val1) && (val1.need_clear == true || ((type == DB_TYPE_VARCHAR || type == DB_TYPE_VARNCHAR) && val1.data.ch.info.need_clear == true))
     {
       pr_clear_value (&val1);
      pr_clear_compressed_string (&val1);
     }

Maybe you should introduce a macro DB_NEED_CLEAR(val) to replace external access of value.need_clear.
Did you consider including the code from pr_clear_compressed_string in pr_clear_value ?